commodore/dependency_templater.py

```diff
@@ -15,12 +16,48 @@
 
 from commodore.config import Config
 from commodore.cruft import create as cruft_create, update as cruft_update
-from commodore.gitrepo import GitRepo, MergeConflict
+from commodore.gitrepo import GitRepo, MergeConflict, default_difffunc
 from commodore.multi_dependency import MultiDependency
 
 SLUG_REGEX = re.compile("^[a-z][a-z0-9-]+[a-z0-9]$")
 
 
+def _ignore_cruft_json_commit_id(
+    before_text: str, after_text: str, fromfile: str = "", tofile: str = ""
+):
+    """Custom diff function which omits `.cruft.json` diffs which only differ in the
+    template commit id."""
+    before_lines = before_text.split("\n")
+    after_lines = after_text.split("\n")
+    diff_lines = difflib.unified_diff(
+        before_lines, after_lines, lineterm="", fromfile=fromfile, tofile=tofile
+    )
+    omit = False
+    if fromfile == ".cruft.json" and tofile == ".cruft.json":
+        # Compute diff without context lines for `.cruft.json` (n=0) and drop the
+        # unified diff header (first 3 lines).
+        minimal_diff = list(
+            difflib.unified_diff(
+                before_lines,
+                after_lines,
+                lineterm="",
+                fromfile=fromfile,
+                tofile=tofile,
+                n=0,
+            )
+        )[3:]
+        # If the context-less diff has exactly 2 lines and both of them start with
+        # '[-+] "commit":', we omit the diff
+        if (
+            len(minimal_diff) == 2
+            and minimal_diff[0].startswith('-  "commit":')
+            and minimal_diff[1].startswith('+  "commit":')
```

Comment:
I suspect it's always guaranteed that the `-` line is before the `+` line?

Reply:
Yes, that's my understanding of the unified diff format returned by `difflib.unified_diff()`